src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/OneVersusAllTrainer.cs

```diff
@@ -227,37 +228,324 @@ public override MulticlassPredictionTransformer<OneVersusAllModelParameters> Fit
+            bool imputeMissingLabelsAsNegative = false,
+            bool useProbabilities = true)
+         : base(env: env, binaryEstimator: binaryEstimator, labelColumnName: labelColumnName, imputeMissingLabelsAsNegative: imputeMissingLabelsAsNegative, useProbabilities: useProbabilities)
+        {
+        }
+
+        private protected override ISingleFeaturePredictionTransformer<TScalarPredictor> TrainOneHelper(IChannel ch,
+            bool useProbabilities, IDataView view, string trainerLabel,
+            ISingleFeaturePredictionTransformer<TScalarPredictor> transformer)
+        {
+            if (useProbabilities)
+            {
+                var calibratedModel = transformer.Model as TDistPredictor;
+
+                // If probabilities are requested and the Predictor is not calibrated or if it doesn't implement the right interface then throw.
+                Host.Check(calibratedModel != null, "Predictor is either not calibrated or does not implement the expected interface");
+
+                // REVIEW: restoring the RoleMappedData, as much as we can.
+                // not having the weight column on the data passed to the TrainCalibrator should be addressed.
+                var trainedData = new RoleMappedData(view, label: trainerLabel, feature: transformer.FeatureColumnName);
+
+                return new BinaryPredictionTransformer<TScalarPredictor>(Host, calibratedModel, trainedData.Data.Schema, transformer.FeatureColumnName);
+            }
+
+            return new BinaryPredictionTransformer<TScalarPredictor>(Host, transformer.Model, view.Schema, transformer.FeatureColumnName);
+        }
+
+        private protected override MulticlassPredictionTransformer<OneVersusAllModelParametersBase<T>> FitHelper(IHost host, bool useProbabilities, TScalarPredictor[] predictors, DataViewSchema schema, string featureColumn, string labelColumnName)
+        {
+            return new MulticlassPredictionTransformer<OneVersusAllModelParametersBase<T>>(Host, OneVersusAllModelParametersBuilder.Create<T>(Host, useProbabilities, predictors.Cast<T>().ToArray()), schema, featureColumn, LabelColumn.Name);
+        }
+
+        private protected override OneVersusAllModelParametersBase<T> TrainCore(IChannel ch, RoleMappedData data, int count)
+        {
+            // Train one-vs-all models.
+            var predictors = new T[count];
+            for (int i = 0; i < predictors.Length; i++)
+            {
+                ch.Info($"Training learner {i}");
+                predictors[i] = (T)TrainOne(ch, Trainer, data, i).Model;
+            }
+            return OneVersusAllModelParametersBuilder.Create<T>(Host, TrainerOptions.UseProbabilities, predictors);
+        }
+    }
+
+    /// <summary>
+    /// Class that holds the static create methods for the <see cref="OneVersusAllModelParameters"/> classes.
+    /// </summary>
+    public class OneVersusAllModelParametersBuilder {
```

Comment:
This class doesn't need to be public. It doesn't contain any public members.

Also, it can be marked as a `static class`.

Reply:
Good points. I will change those.